### PR TITLE
Whishlist model

### LINKED ISSUE #6: There should be a wishlist model
As a user, I want to be able to store items in wishlists

The model should include the following:
- ID (autogenerated by MongoDB/Djongo)
- name: str
- userID: ForeignKey()
- itemIDs: ManyToMany relationship

This will need to be made in the `models.py` file in the /items directory. We'll need to extend the djongo.models.Model class. 
We will also need to create a serializer in a `serializers.py` file in the /items directory to translate back and forth between python-ready and mongoDB-ready data.

### PR DESCRIPTION
**Pull request by FranciscoCamach0**
# Closes #6 

In this pull request, I made a wishlist model that takes in foreign keys (userID, ItemID) from other objects to create a wishlist you can add a list of Items to the wishlist.
### Changes

-  Created wish list model
- creates test cases testing functionality of wishlist model

### Tests 

- Test case for adding items
- Test case for removing items
- Test case for retrieving items/making sure they are there
- Test for erasing items in the wishlist
- Test for if user does't exist wishlist gets deleted

### CODE REVIEW

File: wishlists/serializers.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from rest_framework_mongoengine.serializers import DocumentSerializer
+from .models import Wishlists
+
+class WishlistsSerializer(DocumentSerializer):
+    class Meta:
```
Comment (zliel) [inline wishlists/serializers.py:5]: I believe this should be `model = Wishlists`

File: wishlists/models.py [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
-
+# https://docs.mongoengine.org/tutorial.html#defining-our-documents
+# 1.2.2.4. Handling deletions of references and 1.2.2.1. Posts
+from users.models import User
+from items.models import Items
 # Create your models here.
+class Wishlists(Document):
+    name = StringField(required = True, unique = True, max_length = 30)
+    userID = ReferenceField(User,reverse_delete_rule= CASCADE)
+    item_ids = ListField(ReferenceField(Items),reverse_delete_rule= CASCADE)
```
Comment (zliel) [inline wishlists/models.py]: I think the reverse_delete_rule might need to be in the parentheses for the ReferenceField() part, not the outer ListField part? Not entirely sure

**Review by aimankhan01 — APPROVED** [review]
Looks good

**Review by RutujaN01 — APPROVED** [review]
Looks good.

**Pull request merged**